### PR TITLE
[Dart] Complete function.outter with annotation

### PR DESCRIPTION
**Pull request by GenesisTMS**

### CODE REVIEW

**Review by theHamsta — APPROVED** [review]
Thanks! Could there be annotations be also on other textobjects like classes?

**Comment by GenesisTMS** [comment]
I looked at it more deeply. Fixed some conditional and parameter stuff. Also add annotations to the class you pointed out.
(is it okay to have it in this PR? i can make another since it does more than title says)

Readme updated...
Thanks!

File: queries/dart/textobjects.scm [hunk 1]
```diff
@@ -10,20 +14,21 @@
...
-(function_body) @function.inner
-(type_alias (function_type) @function.inner) @function.outer
+(type_alias (function_type)? @function.inner) @function.outer
 
 ; parameter
-(formal_parameter) @parameter.inner
-(arguments (_) @parameter.inner)
-(normal_parameter_type) @parameter.inner
-(type_parameter) @parameter.inner
+(formal_parameter_list) @parameter.outer
```
Comment (theHamsta) [inline queries/dart/textobjects.scm:24]: parameter.outer is the parameter plus the following comma.

File: queries/dart/textobjects.scm [hunk 2]
```diff
@@ -1,6 +1,10 @@
 ; class
-(class_definition
-  body: (class_body) @class.inner) @class.outer
+((
+  [(marker_annotation)? (annotation)?] @class.outer.start .
+  (class_definition 
+    body: (class_body) @_end @class.inner) @_start
+)
+(make-range! "class.outer" @_start @_end))
```
Comment (theHamsta) [inline queries/dart/textobjects.scm:7]: Thanks for adding this.

File: queries/dart/textobjects.scm (hunk 1 above)
Comment (GenesisTMS) [reply queries/dart/textobjects.scm:24]: Ah, so unless there is a way how to select specific character after parameter (i don't know) this cannot be done with current dart treesitter parser.
Removing outer ones...

Comment (GenesisTMS) [reply queries/dart/textobjects.scm]: Ok i found implementation i `c` textobjects. So i added `parameter.outer`.

**Review by theHamsta — APPROVED** [review]

**Pull request merged**